evennia/utils/ansi.py

```diff
@@ -459,13 +462,18 @@ def parse_ansi(self, string, strip_ansi=False, xterm256=False, mxp=False):
 
         # check cached parsings
         global _PARSE_CACHE
-        cachekey = "%s-%s-%s-%s" % (string, strip_ansi, xterm256, mxp)
+        cachekey = f"{string}-{strip_ansi}-{xterm256}-{mxp}-{truecolor}"
+
         if cachekey in _PARSE_CACHE:
             return _PARSE_CACHE[cachekey]
 
         # pre-convert bright colors to xterm256 color tags
         string = self.brightbg_sub.sub(self.sub_brightbg, string)
 
+        def do_truecolor(part: re.Match, truecolor=truecolor):
+            hex2truecolor = HexColors()
```

Comment:
Why initialize this every regex query? It seems needlessly expensive. It's a stateless class, it could be initialized once at the top of this module.

Reply:
🤦‍♂️ Mistake on my part! Sorry.